app/services/azure-search-service.js

```diff
@@ -0,0 +1,54 @@
+const { SearchClient, AzureKeyCredential } = require('@azure/search-documents')
+const config = require('../config')
+
+/**
+ *
+ * @param {{ chunk_id: string, parent_id: string, chunk: string, title: string, grant_scheme_name: string, source_url: string, content_vector: number[] }} document
+ * @param {SearchClient} searchClient
+ */
+const uploadDocument = async (document, searchClient) => {
+  const uploadResult = await searchClient.uploadDocuments([document])
+
+  const uploadedKeys = uploadResult.results
+    .filter((result) => result.succeeded)
```

Comment:
can any item be null or undefined in this array? if so, the filter will throw an error.
If not, then no worries really.
Same trycatch comment here if you think it would be beneficial

Reply:
I don't think these can be null/undefined